parlai/core/torch_agent.py

```diff
@@ -304,22 +362,23 @@ def batchify(self, obs_batch, sort=False,
                 x_text = [x_text[k] for k in ind_sorted]
 
             xs = torch.LongTensor(len(exs), max(x_lens)).fill_(self.NULL_IDX)
-            if self.use_cuda:
-                xs = xs.cuda()
             for i, ex in enumerate(x_text):
                 xs[i, :ex.shape[0]] = ex
+            if self.use_cuda:
```

Comment:
I'm actually not certain. copies between cuda tensors are faster than copies between cpu tensors, so this might actually be slower in some settings. however, this behavior is useful because then the pytorch teacher which can cache these tensors to disk doesn't have to worry about whether they are cpu or cuda tensors.

Reply:
it might be worth running the speed comparison but I'm not up for doing that this week before I head out on PTO